easyunfold/procar.py

```diff
@@ -32,85 +38,198 @@ def __init__(self, fobj_or_path=None, is_soc=False):
+
+        self.nspins = proj_data.shape[0] // (self.nion * nbands * nkpts)
         self.nspins //= 4 if self._is_soc else 1
 
         # Reshape
-        self.occs.resize((self.nspins, self.nkpts, self.nbands))
-        self.kvecs.resize((self.nspins, self.nkpts, 3))
-        self.kweights.resize((self.nspins, self.nkpts))
-        self.eigenvalues.resize((self.nspins, self.nkpts, self.nbands))
+        occs.resize((self.nspins, nkpts, nbands))
+        kvecs.resize((self.nspins, nkpts, 3))
+        kweights.resize((self.nspins, nkpts))
+        eigenvalues.resize((self.nspins, nkpts, nbands))
 
         # Reshape the array
         if self._is_soc is False:
-            self.proj_data = proj_data.reshape((self.nspins, self.nkpts, self.nbands, self.nion, nproj))
+            proj_data = proj_data.reshape((self.nspins, nkpts, nbands, self.nion, len(self.proj_names)))
+            proj_xyz = None
         else:
-            self.proj_data = proj_data.reshape((self.nspins, self.nkpts, self.nbands, 4, self.nion, nproj))
+            proj_data = proj_data.reshape((self.nspins, nkpts, nbands, 4, self.nion, len(self.proj_names)))
             # Split the data into xyz projection and total
-            self.proj_xyz = self.proj_data[:, :, :, 1:, :, :]
-            self.proj_data = self.proj_data[:, :, :, 0, :, :]
+            proj_xyz = proj_data[:, :, :, 1:, :, :]
+            proj_data = proj_data[:, :, :, 0, :, :]
+
+        # normalise: (for each nspin, nkpt, nband, the sum of the projections over nion and proj_names should be 1)
+        proj_sum = np.sum(proj_data, axis=(-2, -1), keepdims=True)
+        proj_sum[proj_sum == 0] = 1  # just in case, avoid division by zero
+        proj_data /= proj_sum
+
+        if proj_xyz is not None:
+            proj_sum = np.sum(proj_xyz, axis=(-3, -2, -1), keepdims=True)
+            proj_sum[proj_sum == 0] = 1
+            proj_xyz /= proj_sum
+
+        return self.nspins, occs, kvecs, kweights, eigenvalues, proj_data, proj_xyz, parsed_kpoints
+
+    def _read_header_nion_proj_names(self, fobj):
+        """Read the header, nion and proj_names from the PROCAR"""
+        fobj.seek(0)
+        self.header = fobj.readline()
+        # Read the NK, NB and NIONS that are integers
+        _nkpts, _nbands, self.nion = [int(token) for token in re.sub(r'[^0-9]', ' ', fobj.readline()).split()]
+        self.proj_names = None  # projection names
+
+        for line in fobj:
+            if re.match(r'^ion.*tot', line):  # only the first "ion" line, in case of LORBIT >= 12
+                self.proj_names = line.strip().split()[1:-1]
+                break
+
+    def read(self, fobjs_or_paths):
+        """Read and amalgamate the data from a list of PROCARs"""
+
+        def open_file(fobj_or_path):
+            if isinstance(fobj_or_path, (str, Path)):
+                return open(fobj_or_path, encoding='utf-8')  # closed later
+            return fobj_or_path  # already a file-like object, just return it
```

Comment:
```suggestion
            if isinstance(fobj_or_path, (str, Path)):
                return open(fobj_or_path, encoding='utf-8'), True # closed later
            return fobj_or_path, False  # already a file-like object, just return it
```
Can record if the file object is opened by the function or not.